src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,292 @@
+        System.out.println(DELETE_COMMAND + tasks.get(taskPosition));
+        return tasks.removeTask(taskPosition);
+    }
+    /**
+     * Process the user's input deletion of a task. It will display the command that the
+     * task has been deleted, and show the current list of tasks
+     * @param taskPosition The index of the task to delete, starting from 1
+     * @param tasks The list of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the corresponding task deleted
+     */
+    static TaskList<Task> delete(Scanner sc, TaskList<Task> tasks) {
+        int taskPosition = sc.nextInt() - DECREMENT;
+        System.out.println(DELETE_COMMAND + tasks.get(taskPosition));
+        return tasks.removeTask(taskPosition);
+    }
+
+    /*
+    static TaskList<Task> delete(String command, TaskList<Task> tasks, boolean isUi) {
+        int taskPosition = Integer.parseInt(command.substring(Math.max(command.length() - 1, 0))) - DECREMENT;
+        System.out.println(DELETE_COMMAND + tasks.get(taskPosition));
+        return tasks.removeTask(taskPosition);
+    }
+    */
+    /**
+     * Add a new Todo task into the list of tasks from the saved data.
+     * @param description Title of the task
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> toDo(String description, TaskList<Task> tasks) {
+        if (description.trim().length() == 0) {
+            throw new DukeException("Todo must not be empty");
+        }
+        Task newTask = new Todos(description);
+        return tasks.add(newTask);
+    }
+    /**
+     * Add a new Todo task into the list of tasks from the user's input.
+     * @param sc Scanner object to register user's input
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> toDo(Scanner sc, TaskList<Task> tasks) {
+        String description = sc.nextLine();
+        if (description.trim().length() == 0) {
+            throw new DukeException("Todo must not be empty");
+        }
+        Task newTask = new Todos(description);
+        return tasks.add(newTask);
+    }
+
+    /**
+     * Add a new Deadline task into the list of tasks from the saved data.
+     * @param description Title of the task
+     * @param date End date of deadline task
+     * @param tasks List of tasks
+     * @param Type of element stored in the TaskList
+     * @return TaskList New list of tasks with the latest being the todo task added
+     * @exception DukeException
+     */
+    static TaskList<Task> deadline(String description, String date, TaskList<Task> tasks) {
```

Comment:
A more intuitive name would be better here too
```suggestion
    static TaskList<Task> addDeadline(String description, String date, TaskList<Task> tasks) {
```